doc/adrs/088_add_support_for_compartment_expand-collapse.adoc

```diff
@@ -0,0 +1,170 @@
+
+NOTE: This new attribute will not be sent to the frontend through the graphql API.
+
+The action to collapse a node may require to remove the custom property `size` to force the collapsed node to take its minimal space on next layout.
+
+During the node rendering we will add the `collapse` variable to the node variable manager whether the node is already collapsed or if a collapse event has been emit by the user.
+It could be used, for example, in a conditional style expression to display the node with a different node style wheter the node is collpased or not.
+
+==== Collapse/Expand action
+
+We will add a tool only for metadata, like the direct-edit tool or the delete tool in `ViewToolSectionsProvider#createExtraToolSections`:
+
+[source,java]
+----
+
+private List<ToolSection> createExtraToolSections(Object diagramElementDescription, Object diagramElement) {
+    if (diagramElementDescription instanceof NodeDescription nodeDescription) {
+        if (nodeDescription.isCollapsable() && diagramElement instanceof Node node) {
+            List<ITool> collapsingTools = new ArrayList<>();
+            // Collapse expand
+            SingleClickOnDiagramElementTool collapseTool = SingleClickOnDiagramElementTool.newSingleClickOnDiagramElementTool("collapse")
+                    .label("Collapse")
+                    .imageURL(DiagramImageConstants.COLLAPSE_SVG)
+                    .targetDescriptions(targetDescriptions)
+                    .handler(fakeHandler)
+                    .appliesToDiagramRoot(false)
+                    .build();
+            SingleClickOnDiagramElementTool expandTool = SingleClickOnDiagramElementTool.newSingleClickOnDiagramElementTool("expand")
+                    .label("Expand")
+                    .imageURL(DiagramImageConstants.EXPAND_SVG)
+                    .targetDescriptions(targetDescriptions)
+                    .handler(fakeHandler)
+                    .appliesToDiagramRoot(false)
+                    .build();
+            switch(node.getCollapsingState()) {
+                case CollapsingState.expanded:
+                    collapsingTools.add(expandTool);
+                    break;
+                case CollapsingState.collapsed:
+                    collapsingTools.add(collapseTool);
+                    break;
+                default:
+                    break;
+            }
+            var expandCollapseToolSection = ToolSection.newToolSection("expand-collapse-section")
+                    .label("")
+                    .imageURL("")
+                    .tools(collapsingTools)
+                    .build();
+            extraToolSections.add(expandCollapseToolSection);
+        }
+    }
+}
+
+----
+
+To handle the action we will add a new concepts to out api
+
+===== Graphql API
+
+[source,graphql]
+----
+
+extend type Mutation {
+    collapseExpand(input: CollapseExpandInput!): CollapseExpandPayload!
```

Comment:
I would change the name for something like `updateCollapsibleState` or something like that instead because `collapseExpand` may not be the best name